app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -124,29 +121,37 @@ public boolean onTouchEvent(MotionEvent event) {
         return super.onTouchEvent(event);
     }
 
-    private void checkTimePoint() {
-        if (levelData.isEmpty()) {
-            // When the obstacles end - the level is considered passed.
-            winGame();
-            return;
-        }
-
-        if (intervalTimePoint == levelData.getCurrentTimeInterval()) {
-            //  This is just an example of how we can get
-            //  info about an obstacle that should appear at the moment.
-            ObstacleType newObstacleType = levelData.getNewObstacleType();
-            intervalTimePoint = GameConstants.INTERVAL_START_TIME;
-        }
-    }
-
     private void updateGameState() {
         checkTimePoint();
         updateDroidCoordinates();
         updateObstaclesCoordinates();
         updatePlatformCoordinates();
-        // Level Finishing
-        if (timePoint == levelTimePoints) {
-            winGame();
+    }
+
+    private void checkTimePoint() {
+        if (level.isEmpty()) {
+            // When the obstacles end - the level is considered passed.
+            if (obstacleList.isEmpty()){
+                winGame();
+            }
+            return;
+        }
+        if (intervalTimePoint == level.getCurrentTimeInterval()) {
+            ObstacleType newObstacleType = level.getNewObstacleType();
+            // Adding new obstacle to game.
+            switch (newObstacleType){
+                case cactus:
+                    obstacleList.add(new Cactus(screenX, screenY - groundHeight, getResources()));
+                    break;
+                case palm:
+                    obstacleList.add(new Palm(screenX, screenY - groundHeight, getResources()));
+                    break;
+                case bat:
+                    // 700 - random value TODO(Max): calculate y coordinate for bat
```

Comment:
separate the comment from the todo:
// 700 - random value
// TODO(Max): ...

Reply:
Done